integrade/api.py

```diff
@@ -37,14 +39,19 @@ def raise_error_for_status(response):
             response_message = 'text_error_message : {}'.format(
                 pformat(response.text))
 
+        error_headers = {}
+        if isinstance(response.request.headers, dict):
```

Comment:
This won't be different from a dict, only on mocking. I would rather update the test to properly mock a response object by having a dict assigned to response.request.headers.

Reply:
Also maybe worth adding a test to check if the response authorization header is being set to 8 * '*'

Reply:
Here are the changes I would have on this PR

```diff
diff --git a/integrade/api.py b/integrade/api.py
index eadd5c3..2697b17 100644
--- a/integrade/api.py
+++ b/integrade/api.py
@@ -39,11 +39,9 @@ def raise_error_for_status(response):
             response_message = 'text_error_message : {}'.format(
                 pformat(response.text))
 
-        error_headers = {}
-        if isinstance(response.request.headers, dict):
-            error_headers = response.request.headers.copy()
-            if error_headers.get(AUTHORIZATION_HEADER) is not None:
-                error_headers[AUTHORIZATION_HEADER] = '*' * 8
+        error_headers = response.request.headers.copy()
+        if AUTHORIZATION_HEADER in error_headers:
+            error_headers[AUTHORIZATION_HEADER] = '*' * 8
         error_msgs += '\n\n'.join(
             [
                 'request path : {}'.format(pformat(
diff --git a/tests/test_api.py b/tests/test_api.py
index 08ec9c1..fb8ef10 100644
--- a/tests/test_api.py
+++ b/tests/test_api.py
@@ -30,7 +30,7 @@ def mock_request():
     mock_request = mock.Mock(
         body='{"Test Body"}',
         path_url='/example/path/',
-        headers='{"Test Header"}',
+        headers={'Authorization': 'authorizationkey'},
         text='Some text',
     )
     return mock_request
@@ -60,6 +60,7 @@ def bad_response_valid_json():
 def bad_response_invalid_json():
     """Return a mock response with a 404 status code but with bad json."""
     mock_response = mock.Mock(status_code=404, text='<No JSON!>')
+    mock_response.request = mock_request()
     mock_response.json = Mock()
     mock_response.json.side_effect = JSONDecodeError(
         msg='no json',
```